test/typings/test-d/kysely-any.test-d.ts

```diff
@@ -36,6 +36,19 @@ async function testKyselyAnySelects(db: Kysely<any>) {
     ])
     .executeTakeFirstOrThrow()
   expectType<{ bar: any; spam: any; baz: 1 }>(r6)
+
+  const table: 'foo' | 'bar' = Math.random() > 0.5 ? 'foo' : 'bar'
```

Comment:
This `Math.random()` tomfoolery is needed because typescript is too smart. Otherwise it'll use the narrowed type based on the value expression.

Reply:
Well, obviously it doesn't need to be `Math.random()` but something that can't be inferred at compile time.